xdsl/transforms/csl_stencil_to_csl_wrapper.py

```diff
@@ -85,14 +85,15 @@ def match_and_rewrite(self, op: func.FuncOp, rewriter: PatternRewriter, /):
                 )
 
             # retrieve z_dim from done_exchange arg[0]
-            if isa(
+            if isattr(
                 field_t := apply_op.done_exchange.block.args[0].type,
-                stencil.StencilType[
-                    TensorType[Attribute] | memref.MemRefType[Attribute]
-                ],
+                stencil.StencilTypeConstr,
+            ) and isattr(
+                el_type := field_t.element_type,
+                AnyTensorTypeConstr | AnyMemRefTypeConstr,
```

Comment:
this is the only kind of functional change, splitting the check into two, it should still be equivalent

Reply:
LGTM, thanks for doing this